### DISCUSSION TITLE
Future Plans for Motor Friction Parameters (friction, damping, rotorInertia, rotorGearRatio, armature)?

### DISCUSSION THREAD

**Discussion by Cfather** [stack-of-tasks/pinocchio#2435] [Q&A] [answered]
Hi, I have one question about how motor friction parameters will be incorporated in pinocchio in the future.
I noticed that there are the following fields stored in pinocchio::model:
 - friction
 - damping
 - armature
 - rotorInertia
 - rotorGearRatio
They are correctly imported from urdf / sdf / MuJoCo XML. But they are not actually being used anywhere in the dynamics algorithms (except armature). I wonder if there will be such support in the future, like how MuJoCo deals with motor friction parameters.
Specifically, I'm imagining the following changes:
1. friction force in forward/inverse dynamics: +/- friction * sign(v) in aba and rnea.
2. damping force in forward/inverse dynamics: +/- damping * v in aba and rnea.
3. motordynamicsRegressor corresponding to these parameters (similar to jointBodyTorqueRegressor) for friction identification.
I'm not sure what to do with rotorInertia or rotorGearRatio for now.

**Comment by jcarpent** [comment, marked as answer]
We are including part of these terms in Simple simulator. The difficulty is at another level.
In simulators, as they work with impulses, it requires to solve a complementary problem.
For algorithms operating at the acceleration level, like RNEA and ABA, this is indeed easier, as the joint velocity is already known.

The first stage will be to fully parse the quantities, as suggested in #2372.
We will then adapt ABA and RNEA accordingly. We can also think to some options to these algorithms to account or not to the mentionned quantities.

**Comment by olivier-stasse** [comment]
@Cfather to complement @jcarpent answer. The rotorInertia and rotorGearRatio is very important when controlling a real robot specially when it has high rotorGearRatio. Compact actuator typically use a rotorGearRatio of 100 up to 250, and the rotorInertia needs to be accounted for.
